python/pyspark/sql/types.py

```diff
@@ -1445,15 +1447,15 @@ def __new__(self, *args, **kwargs):
             raise ValueError("Can not use both args "
                              "and kwargs to create Row")
         if kwargs:
-            # create row objects
+            # create row object from named arguments, order not guaranteed so will be sorted
             names = sorted(kwargs.keys())
```

Comment:
Yea, this sorting stuff can be removed once we drop Python lower than 3.6. However, keeping `__from_dict__` after ser/de might fix another problem such as https://github.com/apache/spark/pull/26118

Reply:
Thanks for pointing that out. 

In my opinion dropping 3.5 support should be followed by removal of sorting by name (from what I've seen this behavior is confusing at best and it will have no use, once 3.5 support is dropped), although that might hard to sell, especially in a minor version. 

Ideally I would see `Row` going away completely, but that's just pie in the sky.

Reply:
Since we're in 3.0, we might have to consider to drop that sorting by name one .. yes.

Reply:
That'd be great, but 3.0 only deprecates, not drops Python 2 support. So the main why sorting was introduced stays valid. 

If [Spark 2.5](http://apache-spark-developers-list.1001551.n3.nabble.com/DISCUSS-Spark-2-5-release-td27963.html) was accepted, Python 2 could be deprecated there, and dropped in 3.0. In general I think that bringing Python 2 into Spark 3 is a mistake, but I am first to admit I am biased.

Reply:
I think Spark 2.5 idea looks almost failed (and I either don't personally support that idea). This thing is rather legitimate to make it happen between minor releases. Actually, 3.0 is getting close. We could just let 3.0 release first and remove that sorting behaviour too optionally.